test/puppetlabs/puppetdb/testutils/services.clj

```diff
@@ -31,6 +35,56 @@
 (def ^:dynamic *extra-log-config* nil)
 (def ^:dynamic *extra-appender-config* nil)
 
+(defmacro with-log-level
+  "Sets the (logback) log level for the logger specified by logger-id
+  during the execution of body.  If logger-id is not a class, it is
+  converted via str, and the level must be a clojure.tools.logging
+  key, i.e. :info, :error, etc."
+  [logger-id level & body]
+  ;; Assumes use of logback (i.e. logger supports Levels).
+  `(let [logger-id# ~logger-id
+         logger-id# (if (class? logger-id#) logger-id# (str logger-id#))
+         logger# (.getLogger (LoggerFactory/getILoggerFactory) logger-id#)
+         original-level# (.getLevel logger#)]
+     (.setLevel logger# (case ~level
+                          :trace Level/TRACE
+                          :debug Level/DEBUG
+                          :info Level/INFO
+                          :warn Level/WARN
+                          :error Level/ERROR
+                          :fatal Level/ERROR))
```

Comment:
Assuming I didn't make a mistake, slf4j doesn't support a fatal level (it's redirected to error) https://github.com/clojure/tools.logging/blob/9b33f19dd07fbf3f1c7f7f8f30ff987b6f51f7cc/src/main/clojure/clojure/tools/logging/impl.clj#L76